### PR TITLE
Expose several GLES3 built-ins to GLES2

### LINKED ISSUE #30940: Porting GLES3 built-in functions to GLES2
**The Problem**
GLES3 supports more built in functions than GLES2 does (like ``round()``, ``isnan()``, ``sinh()``, etc.) . Godot's shading language follows the GLES3 spec so when these functions are used in GLES2, they result in a compile error (see #24850). Our current solution is to track these differences in the [Differences between GLES2 and GLES3 tutorial](http://docs.godotengine.org/en/latest/tutorials/misc/gles2_gles3_differences.html#shading-language), however, that hasn't been enough. 

**The Solution**
Some of these functions can be implemented in glsl. For example ``round()`` can be implemented as:

```
float round(float x) {
    return floor(x + 0.5);
}
```

Accordingly, someone (maybe YOU!) needs to go through and determine which, if any, other functions can be implemented easily in glsl and then implement them. 

In implementing them, you can just follow how it was implemented for ``round()`` in 
https://github.com/godotengine/godot/pull/30895

### PR DESCRIPTION
**Pull request by Chaosus**
Its implementation of several functions for GLES2 in Spatial and CanvasItem shaders.

- sinh
- cosh
- tanh
- asinh
- acosh
- atanh
- determinant
- round*
- roundEven
- isnan
- isinf
- outerProduct
- transpose
- trunc

(*)for Spatial - its already been exposed for Canvas in #30895)

Significant step to close #30940

inverse(mat4) is too hard for me to implement - maybe somebody other can make it !

### CODE REVIEW

**Comment by clayjohn** [comment]
``transpose()`` and ``outerproduct()`` should be wrapped in an ``#ifndef GLES_OVER_GL`` statement as they are both implemented for desktop GL but not for GL ES 2.

**Comment by Chaosus** [comment]
@clayjohn Fixed ^^

File: drivers/gles2/shaders/stdlib.glsl [hunk 1]
```diff
@@ -51,14 +183,163 @@ highp vec3 round(highp vec3 x) {
...
+}
+
+highp mat3 transpose(highp mat3 m) {
+	return mat3(
+			vec3(m[0].x, m[1].x, m[2].x),
+			vec3(m[0].y, m[1].y, m[2].y),
+			vec3(m[0].z, m[1].z, m[2].z));
+}
+
+highp mat4 transpose(highp mat4 m) {
```
Comment (clayjohn) [inline drivers/gles2/shaders/stdlib.glsl]: ``transpose(mat4)`` is used inside canvas.glsl and scene.glsl, so it should be exposed outside the ``#ifdef TRANSPOSE_USED`` but inside the ``#ifndef GLES_OVER_GL``. Sorry I missed this before.

Comment (Chaosus) [reply drivers/gles2/shaders/stdlib.glsl]: No problem - fixed

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**